packages/fxa-auth-server/lib/senders/select_email_services.js

```diff
@@ -94,115 +163,60 @@ module.exports = (log, config, mailer, emailService) => {
     }
 
     if (liveConfig) {
-      try {
-        liveConfig = JSON.parse(liveConfig);
-      } catch (err) {
-        log.error('emailConfig.parse.error', { err: err.message });
+      if (liveConfig === lastSeenLiveConfig) {
+        liveConfig = lastSeenLiveConfigParsed;
+      } else {
+        try {
+          lastSeenLiveConfig = liveConfig;
+          liveConfig = LiveConfig.parse(liveConfig);
+          lastSeenLiveConfigParsed = liveConfig;
+        } catch (err) {
+          lastSeenLiveConfig = lastSeenLiveConfigParsed = null;
+          log.error('emailConfig.parse.error', { err: err.message });
+        }
       }
     }
 
-    const services = await emailAddresses.reduce(
-      async (promise, emailAddress) => {
```

Comment:
Not sad to see the end of this very clever, but terrifying nested-promise-reducer-loop. Being able to test the regexes synchronously is a nice side benefit of this change.